pkg/securelaunch/launcher/launcher.go

```diff
@@ -0,0 +1,87 @@
+package launcher
+
+import (
+	"fmt"
+	"io"
+	"log"
+
+	"github.com/u-root/u-root/pkg/boot"
+	"github.com/u-root/u-root/pkg/kexec"
+	slaunch "github.com/u-root/u-root/pkg/securelaunch"
+	"github.com/u-root/u-root/pkg/securelaunch/measurement"
+	"github.com/u-root/u-root/pkg/uio"
+)
+
+/* describes the "launcher" section of policy file */
+type Launcher struct {
+	Type   string            `json:"type"`
+	Params map[string]string `json:"params"`
+}
+
+/*
+ * This function boots the target kernel based on information provided
+ * in the "launcher" section of policy file.
+ * Boot fn,
+ * - extracts the kernel, initrd and cmdline from the "launcher" section of policy file.
+ * - measures the kernel and initrd file into the tpmDev (tpm device).
+ * - mounts the disks where the kernel and initrd file are located.
+ * - uses kexec to boot into the target kernel.
+ * returns error
+ * - if measurement of kernel and initrd fails
+ * - if mount fails
+ * - if kexec fails
+ */
+func (l *Launcher) Boot(tpmDev io.ReadWriteCloser) error {
+
+	if l.Type != "kexec" {
+		log.Printf("launcher: Unsupported launcher type. Exiting.")
+		return fmt.Errorf("launcher: Unsupported launcher type. Exiting")
+	}
+
+	slaunch.Debug("Identified Launcher Type = Kexec\n")
+
+	// TODO: if kernel and initrd are on different devices.
+	kernel := l.Params["kernel"]
+	initrd := l.Params["initrd"]
+	cmdline := l.Params["cmdline"]
+
+	slaunch.Debug("********Step 6: Measuring kernel, initrd ********\n")
+	if e := measurement.HashFile(tpmDev, kernel); e != nil {
```

Comment:
I'm wondering if you dedup GetMountedFilePath here -- 

if you move the slaunch.GetMountedFilePath to before the measurement, then you could just pass the path to measurement.HashFile and just hash a file system path?

that way HashFile wouldn't have to call GetMountedFilePath as well? i'm not sure where else it's used yet.

Reply:
GetMountedFile path is used by every collector and launcher and eventlog sub-pkg.
With every collector (dmi, cpuid, files, storage) , there is an opportunity for user to enter a path in policy file. Not to mention every future collector.will need this too..
```
pkg/securelaunch/policy/policy.go
51:	mntFilePath, mountPath, e := slaunch.GetMountedFilePath(val, false) // false means readonly mount

pkg/securelaunch/measurement/files.go
48:	mntFilePath, mountPath, e := slaunch.GetMountedFilePath(inputVal, false) // false means readonly mount

pkg/securelaunch/measurement/cpuid.go
82:	filePath, mountPath, r := slaunch.GetMountedFilePath(cpuidTargetPath, true) // true = rw mount option

pkg/securelaunch/eventlog/eventlog.go
72:	filePath, mountPath, r := slaunch.GetMountedFilePath(eventlogPath, true) // true = rw mount option

pkg/securelaunch/launcher/launcher.go
59:	k, _, e := slaunch.GetMountedFilePath(kernel, false) // false=read only mount option
65:	i, _, e := slaunch.GetMountedFilePath(initrd, false) // false=read only mount option
```